Ska/engarchive/derived/thermal.py

```diff
@@ -11,6 +11,21 @@
 class DerivedParameterThermal(base.DerivedParameter):
     content_root = 'thermal'
 
+    def fix_bus_5_and_bus_6(self, data):
+        # These zones cannot be commanded if Bus 5 or Bus 6 is disabled
+        bus_5_and_6_zones = ['4OHTRZ27', '4OHTRZ28', '4OHTRZ43', '4OHTRZ45',
+                             '4OHTRZ47', '4OHTRZ49', '4OHTRZ52', '4OHTRZ53',
+                             '4OHTRZ29', '4OHTRZ35', '4OHTRZ42', '4OHTRZ44',
+                             '4OHTRZ46', '4OHTRZ48', '4OHTRZ50', '4OHTRZ54']
+
+        for msid in bus_5_and_6_zones:
+            if msid in data.keys():
+                data[msid].vals = (data['4S1PWR05'].vals == 'ENAB') & \
+                                  (data['4S1PWR06'].vals == 'ENAB') & \
+                                  (data[msid].vals == 1)
+                # Convert bool type back to original int8 dtype
+                data[msid].vals = data[msid].vals.astype(np.int8)
```

Comment:
```suggestion
               # Heater must be off if either bus is disabled
                disa = (data['4S1PWR05'].vals == 'DISA') | (data['4S1PWR06'].vals == 'DISA')
                data[msid].vals[disa] = 0
```

Reply:
That looks good to me. Instead of _redefining_ the `data[msid].vals` array so that each element is True when the expressions in lines 23-24 all evaluate to true, you are suggesting that the existing `data[msid].vals` array should be modified in place.